x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecks.java

```diff
@@ -116,4 +119,20 @@ static DeprecationIssue indexNameCheck(IndexMetaData indexMetaData) {
         }
         return null;
     }
+
+    static DeprecationIssue nodeLeftDelayedTimeCheck(IndexMetaData indexMetaData) {
+        String setting = UnassignedInfo.INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.getKey();
+        String value = indexMetaData.getSettings().get(setting);
+        if (Strings.isNullOrEmpty(value) == false) {
+            TimeValue parsedValue = TimeValue.parseTimeValue(value, setting);
+            if (parsedValue.getNanos() < 0) {
+                return new DeprecationIssue(DeprecationIssue.Level.WARNING,
+                    "Negative values for index.unassigned.node_left.delayed_timeout are deprecated and should be set to 0",
```

Comment:
Please could you use `setting.getKey()` instead of the literal `"index.unassigned.node_left.delayed_timeout"`?

Reply:
done, thanks!